<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1235</width>
    <height>756</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: transparent;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLineEdit" name="chooseStart">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>70</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Sitka Text</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLineEdit
{
	background-color: white;
	color:black;
	border-style: outset;
	border-width: 2px;
	border-radius:5px;
	border-color:gray;
}
QLineEdit::Focus
{
	border-width: 3.5px;
}

</string>
    </property>
    <property name="placeholderText">
     <string>  Choose start location</string>
    </property>
   </widget>
   <widget class="customButton" name="switchbutton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>140</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Swap start location and destination</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: transparent;
    border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}


QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/switch.svg</normaloff>:/images/switch.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>30</width>
      <height>30</height>
     </size>
    </property>
   </widget>
   <widget class="customButton" name="spbutton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>220</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Sitka Text</family>
      <pointsize>12</pointsize>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="toolTip">
     <string>Return 5 shortest paths</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.2);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string>Top shortest paths</string>
    </property>
   </widget>
   <widget class="customMap" name="map">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>30</y>
      <width>791</width>
      <height>671</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>2</horstretch>
      <verstretch>2</verstretch>
     </sizepolicy>
    </property>
    <property name="toolTipDuration">
     <number>1</number>
    </property>
   </widget>
   <widget class="customButton" name="flowbutton">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>220</y>
      <width>171</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Sitka Text</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>Return maximum flow paths</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.2);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string>Maximum flow</string>
    </property>
   </widget>
   <widget class="customButton" name="resetbutton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>70</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Reset action</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: transparent;
    border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}


QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/reset.svg</normaloff>:/images/reset.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>30</width>
      <height>30</height>
     </size>
    </property>
   </widget>
   <widget class="QLineEdit" name="chooseDes">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>140</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Sitka Text</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLineEdit
{
	background-color: white;
	color:black;
	border-style: outset;
	border-width: 2px;
	border-radius:5px;
	border-color:gray;
}
QLineEdit::Focus
{
	border-width: 3.5px;
}

</string>
    </property>
    <property name="placeholderText">
     <string>  Choose destination</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="outText">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>330</y>
      <width>301</width>
      <height>331</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QTextEdit
{
	border: none;
}

QTextEdit::hover
{
	border: 1px solid black;
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Sitka Small'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="customButton" name="buttonStart">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>70</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Pick start location</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: transparent;
    border-style: offset;
	border-radius: 5px;
	border: 2px solid black;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}

QPushButton:focus {
    background-color: rgba(0, 0, 0, 0.2);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/picker.svg</normaloff>:/images/picker.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>30</width>
      <height>30</height>
     </size>
    </property>
   </widget>
   <widget class="customButton" name="buttonEnd">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>140</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Pick destination</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: transparent;
    border-style: offset;
	border-radius: 5px;
	border: 2px solid black;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}

QPushButton:focus {
    background-color: rgba(0, 0, 0, 0.2);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/picker.svg</normaloff>:/images/picker.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>30</width>
      <height>30</height>
     </size>
    </property>
   </widget>
   <widget class="customButton" name="nextbutton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>270</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Navigate right</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	background-color: transparent;
	border: none;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}


QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/next.svg</normaloff>:/images/next.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>40</width>
      <height>40</height>
     </size>
    </property>
   </widget>
   <widget class="customButton" name="prevbutton">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>270</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Navigate left</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	background-color: transparent;
	border: none;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.1);
}


QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/prev.svg</normaloff>:/images/prev.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>40</width>
      <height>40</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="labelPath">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>260</y>
      <width>331</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Tap above to see the result</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="startlabel">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>110</y>
      <width>291</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Please select start location.</string>
    </property>
   </widget>
   <widget class="QLabel" name="endlabel">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>180</y>
      <width>291</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Please select destination.</string>
    </property>
   </widget>
   <widget class="QLabel" name="startIcon">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>70</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap resource="resource.qrc">:/images/start.svg</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="endIcon">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>140</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap resource="resource.qrc">:/images/des.svg</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="mapToggle">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>50</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Toggle map on</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: rgba(0, 0, 0, 0.1);
	border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.04);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/mapToggle.svg</normaloff>:/images/mapToggle.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>50</width>
      <height>50</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="graph">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Toggle full graph</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: rgba(0, 0, 0, 0.1);
	border: none;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: rgba(0, 0, 0, 0.04);
}

QPushButton:pressed {
    background-color: rgba(0, 0, 0, 0.4);
	text-decoration: underline;
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/images/graph.svg</normaloff>:/images/graph.svg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>35</width>
      <height>35</height>
     </size>
    </property>
   </widget>
   <zorder>labelPath</zorder>
   <zorder>outText</zorder>
   <zorder>chooseStart</zorder>
   <zorder>switchbutton</zorder>
   <zorder>spbutton</zorder>
   <zorder>map</zorder>
   <zorder>flowbutton</zorder>
   <zorder>resetbutton</zorder>
   <zorder>chooseDes</zorder>
   <zorder>buttonStart</zorder>
   <zorder>buttonEnd</zorder>
   <zorder>nextbutton</zorder>
   <zorder>prevbutton</zorder>
   <zorder>startlabel</zorder>
   <zorder>endlabel</zorder>
   <zorder>startIcon</zorder>
   <zorder>endIcon</zorder>
   <zorder>mapToggle</zorder>
   <zorder>graph</zorder>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>customMap</class>
   <extends>QGraphicsView</extends>
   <header>customgraphics.h</header>
  </customwidget>
  <customwidget>
   <class>customButton</class>
   <extends>QPushButton</extends>
   <header>customgraphics.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
